### PR TITLE
Add v2 to go.mod

### PR DESCRIPTION
**Pull request by pr0n00gler**

### CODE REVIEW

File: app/upgrades/v2.0.0/upgrades_test.go [hunk 1]
```diff
@@ -24,8 +24,8 @@ import (
 	globalfeetypes "github.com/cosmos/gaia/v11/x/globalfee/types"
 	"github.com/stretchr/testify/suite"
 
-	v110 "github.com/neutron-org/neutron/app/upgrades/v2.0.0"
-	"github.com/neutron-org/neutron/testutil"
+	v110 "github.com/neutron-org/neutron/v2/app/upgrades/v2.0.0"
```
Comment (swelf19) [inline app/upgrades/v2.0.0/upgrades_test.go]: pls change import alias

**Comment by pr0n00gler** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/7085807788

File: testutil/test_helpers.go [hunk 2]
```diff
@@ -282,8 +282,8 @@ func (suite *IBCConnectionTestSuite) GetNeutronZoneApp(chain *ibctesting.TestCha
 }
 
 func (suite *IBCConnectionTestSuite) StoreTestCode(ctx sdk.Context, addr sdk.AccAddress, path string) uint64 {
-	// wasm file built with https://github.com/neutron-org/neutron-contracts/tree/main/contracts/reflect
-	// wasm file built with https://github.com/neutron-org/neutron-dev/tree/feat/ica-register-fee-update/contracts/neutron_interchain_txs
+	// wasm file built with https://github.com/neutron-org/neutron/v2-contracts/tree/main/contracts/reflect
+	// wasm file built with https://github.com/neutron-org/neutron/v2-dev/tree/feat/ica-register-fee-update/contracts/neutron_interchain_txs
```
Comment (swelf19) [inline testutil/test_helpers.go]: not necessary to rename

File: wasmbinding/bindings/msg.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ type ProtobufAny struct {
 }
 
 // NeutronMsg is used like a sum type to hold one of custom Neutron messages.
-// Follow https://github.com/neutron-org/neutron-sdk/blob/main/packages/neutron-sdk/src/bindings/msg.rs
+// Follow https://github.com/neutron-org/neutron/v2-sdk/blob/main/packages/neutron-sdk/src/bindings/msg.rs
```
Comment (swelf19) [inline wasmbinding/bindings/msg.go]: not necessary to rename

File: x/contractmanager/keeper/failure.go [hunk 4]
```diff
@@ -19,7 +19,7 @@ import (
 // must be OS/library version agnostic and carry a concrete defined error message. One of the good
 // ways to do so is to redact error using the RedactError func as it is done in SudoLimitWrapper
 // Sudo method:
-// https://github.com/neutron-org/neutron/blob/eb8b5ae50907439ff9af0527a42ef0cb448a78b5/x/contractmanager/ibc_middleware.go#L42.
+// https://github.com/neutron-org/neutron/v2/blob/eb8b5ae50907439ff9af0527a42ef0cb448a78b5/x/contractmanager/ibc_middleware.go#L42.
```
Comment (swelf19) [inline x/contractmanager/keeper/failure.go]: wrong link format

Comment (mmikev) [reply x/contractmanager/keeper/failure.go]: fixed

File: wasmbinding/bindings/msg.go (hunk 3 above)
Comment (mmikev) [reply wasmbinding/bindings/msg.go]: fixed

File: testutil/test_helpers.go (hunk 2 above)
Comment (mmikev) [reply testutil/test_helpers.go]: fixed

File: app/upgrades/v2.0.0/upgrades_test.go (hunk 1 above)
Comment (mmikev) [reply app/upgrades/v2.0.0/upgrades_test.go]: fixed

**Review by swelf19 — APPROVED** [review]

**Pull request merged**